thinc/model.py

```diff
@@ -22,11 +23,18 @@ def empty_init(model: "Model", *args, **kwargs) -> "Model":
     return model
 
 
+class ModelThreadState(threading.local):
+    operators: Dict[str, Callable[["Model", "Model"], "Model"]]
+
+    def __init__(self):
+        self.operators = {}
+
+
 class Model(Generic[InT, OutT]):
     """Class for implementing Thinc models and layers."""
 
     global_id: int = 0
```

Comment:
This ID is maybe a problem...We want this to be *properly* global, it will cause problems if models come back from other threads and IDs aren't unique. Should we use a lock here?

Reply:
Sure, I'll add a threading lock and acquire it when incrementing the global value